service/share/share_test.go

```diff
@@ -28,3 +29,30 @@ func TestGetShare(t *testing.T) {
 	err = serv.Stop(ctx)
 	require.NoError(t, err)
 }
+
+func TestService_GetSharesByNamespace(t *testing.T) {
+	var tests = []struct {
+		amountShares       int
+		expectedShareCount int
+	}{
+		{amountShares: 4, expectedShareCount: 1},
+		{amountShares: 16, expectedShareCount: 2},
+		{amountShares: 128, expectedShareCount: 1},
+	}
+
+	for i, tt := range tests {
+		t.Run(strconv.Itoa(i), func(t *testing.T) {
+			serv, root := RandServiceWithTree(t, tt.amountShares)
+			randNID := root.RowsRoots[(len(root.RowsRoots)-1)/2][:8]
+			if tt.expectedShareCount > 1 {
+				// make it so that two rows have the same namespace ID
+				root.RowsRoots[(len(root.RowsRoots) / 2)] = root.RowsRoots[(len(root.RowsRoots)-1)/2]
+			}
+
+			shares, err := serv.GetSharesByNamespace(context.Background(), root, randNID)
+			require.NoError(t, err)
```

Comment:
Need test cases for the not-found-cases here as well. Can also happen in a follow-up PR if there is an issue for that.

Reply:
follow-up PR for edge case testing

Reply:
https://github.com/celestiaorg/celestia-node/issues/178